internal/pkg/service/buffer/watcher/apinode/state.go

```diff
@@ -93,63 +92,84 @@ func (s *state) GetReceiver(receiverKey key.ReceiverKey) (out ReceiverCore, foun
 	return out, true, unlockFn
 }
 
-func (s *state) onError(err error) {
-	s.logger.Error(err)
-}
-
 // The function belongs to the state struct, but generic method cannot be currently defined.
 func watch[T fmt.Stringer](ctx context.Context, wg *sync.WaitGroup, s *state, prefix etcdop.PrefixT[T], rev *revision.Syncer) *stateOf[T] {
 	tree := prefixtree.New[T]()
 
 	initDone := make(chan error)
-	ch := prefix.GetAllAndWatch(ctx, s.client, s.onError, etcd.WithCreatedNotify(), etcd.WithPrevKV())
-
-	// Log only changes, not initial load
-	logsEnabled := atomic.NewBool(false)
+	ch := prefix.GetAllAndWatch(ctx, s.client, etcd.WithCreatedNotify(), etcd.WithPrevKV())
 
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
 
-		// Channel is closed on shutdown, so the context does not have to be checked
-		for events := range ch {
-			if err := events.InitErr; err != nil {
-				initDone <- err
+		// Reset the tree on the restart event.
+		reset := false
```

Comment:
Teraz nasleduje 3x uprava logiky v apke, v kazdom `watcher`:
- Implementacia ide az potom, chcem ukazat ako sa to pouziva.
- Namiesto `onError` chodi `resp.Err != nil`, teda vsetko sa deje v jednom `channel`, tym pado sa to jednoduchsie testuje a synchronizuje.
- Pribudol flag `Restarted`, to nastane ak sa narazi na `ErrCompacted`, ako je to vysvetlene v PR desc.
- V tom pripade musime aj na aplikacnej urovni restartovat stav.
- Je dolezite aby sa to stalo atomicky.
- Teda najprv sa vsetko vymaze, a potom sa spracuju `events` z `GetAll` phase, ktore obsahuju aktualny stav.
- Vo vacsine pripadov sa nic nezmeni, a pokracuje sa dalej, ... ale pocas vypadku mohla nastat aj nejaka zmena, a ta je teda teraz uz spravne osetrena.
- Lepsie pozriet asi vysledny kod, ako diff.

Reply:
V tomto commite je fixnuta `watcher.APINode`.